### ISSUE TITLE
Boost 1.80.0 Bind Placeholder warnings

### ISSUE THREAD

**Issue by simenon** [mat007/turtle#114]
We recently upgraded to boost 1.80.0 in combination with the latest turtle library. 

In the compilation of our project we are getting te following warnings with turtle

`/opt/boost_1_80_0/include/boost/bind.hpp:41:1: note: #pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.`

relating to 
```
turtle/detail/action.hpp:19:0,
turtle/detail/function.hpp:18,
turtle/detail/functor.hpp:13,
turtle/reset.hpp:15,
turtle/mock.hpp:17,
```

Will turtle be made compliant with boost 1.80.0 ?

For the moment we define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior

**Comment by Flamefire** [comment]
There likely just needs to be another release of turtle. Try the recent master which has lot's of changes improving compiletimes and more. Note that it also requires C++14

**Comment by Flamefire** [comment]
This is fixed with Turtle 2.0
